components/tcf/firstLayer/src/index.js

```diff
@@ -1,114 +1,204 @@
-import React, {useState, useEffect} from 'react'
+import React, {useState, useEffect, useRef, useCallback} from 'react'
 import PropTypes from 'prop-types'
 import SuiButton from '@s-ui/react-atom-button'
 import SuiModal from '@s-ui/react-molecule-modal'
-import IconClose from './components/IconClose'
-import InfoCard from './components/InfoCard'
+import SuiNotification from '@s-ui/react-molecule-notification'
+import IconClose from './iconClose'
+import {I18N} from './settings'
+
+const CLASS = 'sui-TcfFirstLayer'
 
 export default function TcfFirstLayer({
-  loadUserConsent,
+  lang = 'es',
+  logo,
+  isMobile,
+  isTablet,
   getVendorList,
   saveUserConsent,
-  openSecondLayer
+  openSecondLayer,
+  openCookiepolicyLayer
 }) {
-  const [saveButtonDisabled, setSaveButtonDisabled] = useState(true)
-  const [state, setState] = useState(null)
-  const [vendorListState, setVendorListState] = useState(null)
-  const [modalOpen, setModalOpen] = useState(true)
+  const [show, setShow] = useState(true)
+  const i18n = I18N[lang]
 
-  useEffect(() => {
-    const loadConsent = async () => {
-      const {purpose} = await loadUserConsent()
-      setState({purposes: purpose})
+  const handleCookiePolicyLayerClick = useCallback(() => {
+    cookiesPolicyLink.current &&
+      cookiesPolicyLink.current.removeEventListener(
+        'click',
+        handleCookiePolicyLayerClick
+      )
+    openCookiepolicyLayer()
+  })
+  const cookiesPolicyLink = useRef()
+  const ref = useRef(null)
+  const textRef = useCallback(
+    node => {
+      if (ref.current) {
+        ref.current = null
+      }
+
+      if (node) {
+        cookiesPolicyLink.current = node.querySelector(
+          `.${CLASS}-body-info-link`
+        )
+        if (cookiesPolicyLink) {
+          cookiesPolicyLink.current.addEventListener(
+            'click',
+            handleCookiePolicyLayerClick
+          )
+        }
+        ref.current = node
+      }
+    },
+    [handleCookiePolicyLayerClick]
+  )
+
+  let initialYOffset
+  let count = 0
+  const checkScroll = () => {
+    count++
+    // use counter to avoid initial window.pageYOffset pointing erroneously to 0
+    if (count === 3) {
+      initialYOffset = window.pageYOffset
     }
-    const loadVendorList = async () => {
-      const {purposes} = await getVendorList()
-      setVendorListState({purposes})
+    if (
+      (count >= 3 && initialYOffset - window.pageYOffset >= 250) ||
```

Comment:
if 250 is the IAB requirement for scrolling, maybe it should be a declared `const` instead of a magic number (it's being used at least in 2 lines, that should change if tomorrow the IAB says 300 instead of 250)

Reply:
Done, I don't know if is an IAB requirement or Adevinta requirement, but I've added it as a constant. Thanks